foundation-ui/utils/ReactUtil.js

```diff
@@ -18,16 +31,11 @@ const ReactUtil = {
       return null;
     }
 
-    if (Array.isArray(elements) && elements.length === 1
-      && React.isValidElement(elements[0]) && !alwaysWrap) {
-      return elements[0];
-    }
-
-    if (React.isValidElement(elements) && !alwaysWrap) {
-      return elements;
+    if (shouldWrapElements(elements, alwaysWrap)) {
```

Comment:
We talked about this briefly in person, but I'll respond here for visibility:

I moved this to a separate function solely to help the readability of this method. I'm not married to this approach by any means, so let's discuss a better way to do this!